nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -74,4 +87,16 @@ protected DocumentModel getClientModel(String clientId) {
             return null;
         });
     }
+
+    protected List<DocumentModel> queryClients() {
+        DirectoryService service = Framework.getService(DirectoryService.class);
+        return Framework.doPrivileged(() -> {
+            try (Session session = service.open(OAUTH2CLIENT_DIRECTORY_NAME)) {
+                return session.query(Collections.emptyMap());
+            } catch (DirectoryException e) {
+                log.error("Error while fetching provider directory", e);
+                return Collections.emptyList();
```

Comment:
It's better to throw a `NuxeoException` than return an empty list. If this shouldn't happen then we should throw.